search-api/src/main/kotlin/org/needlehack/searchapi/controller/SearchController.kt

```diff
@@ -1,43 +1,40 @@
 package org.needlehack.searchapi.controller
 
-import org.needlehack.searchapi.model.FeedItem
 import io.searchbox.client.JestClient
 import io.searchbox.core.Search
 import io.searchbox.core.SearchResult
-import org.elasticsearch.index.query.DisMaxQueryBuilder
-import org.elasticsearch.index.query.QueryBuilders
+import org.elasticsearch.index.query.QueryBuilders.disMaxQuery
+import org.elasticsearch.index.query.QueryBuilders.termQuery
 import org.elasticsearch.search.builder.SearchSourceBuilder
-import org.springframework.beans.factory.annotation.Autowired
+import org.needlehack.searchapi.model.FeedItem
+import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
 
 @RestController
-class SearchController {
-
-	@Autowired
-	private val client: JestClient? = null
-
-	@Value("\${jest.elasticsearch.index}")
-	private var indexName: String? = "feed-collector"
 
-	@GetMapping("/search")
-	fun search(@RequestParam(value = "term", defaultValue = "*") term: String): List<SearchResult.Hit<FeedItem, Void>>? {
+class SearchController(val client: JestClient, @Value("\${jest.elasticsearch.index}") val indexName: String) {
 
-		System.out.println("You want to seach: $term")
+    val log = LoggerFactory.getLogger(SearchController::class.java)
 
-		val searchSourceBuilder = SearchSourceBuilder()
-		searchSourceBuilder.query(QueryBuilders.disMaxQuery().add(QueryBuilders.termQuery("topics", term)).add(QueryBuilders.termQuery("title", term)).tieBreaker(0.3f))
+    @GetMapping("/search")
+    fun search(@RequestParam(value = "term", defaultValue = "*") term: String): List<SearchResult.Hit<FeedItem, Void>> {
 
-		val search: Search.Builder = Search.Builder(searchSourceBuilder.toString())
-		if (indexName != null) search.addIndex(indexName)
+        log.info("You want to seach: [$term]")
+        val searchSource = SearchSourceBuilder()
+                .query(disMaxQuery()
+                        .add(termQuery("topics", term))
+                        .add(termQuery("title", term)).tieBreaker(0.3f))
```

Comment:
We'd have to verify if 0.3f is an useful value for our purpose. Information about that: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-dis-max-query.html

Reply:
I agree with you. Maybe we should do research on this topic.